### PR TITLE
Use emacs' customization system for the indentation

### PR DESCRIPTION
**Pull request by martenlienen**
The title says it all.

### CODE REVIEW

**Comment by yuyichao** [comment]
Verified locally that it works (on emacs 24 at least). I'm not a big fan of alternative indentation but I think we can have this since julia is not indentation sensitive.

@Wilfred Does this look to you?

**Comment by yuyichao** [comment]
Would this make the emacs test fail if the variable is not having the default value?

File: contrib/julia-mode.el [hunk 1]
```diff
@@ -42,7 +42,15 @@
...
 (defvar julia-mode-hook nil)
 
-(defvar julia-basic-offset)
+(defgroup julia ()
+  "Major mode for the julia programming language."
+  :group 'languages
+  :prefix "julia-")
+
+(defcustom julia-indent-offset 4
+  "Indentation per level."
```
Comment (Wilfred) [inline contrib/julia-mode.el]: Perhaps say "indentation in spaces" to make the units explicit?

**Comment by Wilfred** [comment]
:+1: looks good to me, and I prefer the new name of the variable. @yuyichao is right that the tests assume a four space indent, perhaps update `julia--should-indent` to set the variable to 4?

**Comment by cqql** [comment]
I wrote this, because I though julia used an indentation of two spaces as in [the example for immutable types](http://julia.readthedocs.org/en/latest/manual/types/#immutable-composite-types). After it was done, I checked the julia source code and saw that you actually use 4, but thought, I should send this patch anyway.

The doc string now reads a little nicer.

File: contrib/julia-mode.el [hunk 2]
```diff
@@ -582,12 +591,12 @@ foo() =
...
      "
 if foo
-        
+
 bar
 end"
      "
 if foo
-    
+
```
Comment (yuyichao) [inline contrib/julia-mode.el]: Were these white space intentional?

Comment (cqql) [inline contrib/julia-mode.el]: Sorry, I actually checked for this, but I must have missed it, when I amended the commits. My whitespace-removal trigger is messing this up.

Comment (cqql) [inline contrib/julia-mode.el]: Fixed.

**Comment by yuyichao** [comment]
Local tests all passes. Thanks.

**Comment by rahuldave** [comment]
D
Sent on my BlackBerry® from Vodafone

-----Original Message-----

**Pull request merged**